android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/models/NativePayment.java

```diff
@@ -0,0 +1,88 @@
+package com.shopify.unity.buy.models;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.shopify.unity.buy.utils.Logger;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * @author Flavio Faria
+ */
+
+public final class NativePayment implements JsonSerializable {
+
+    private static final String BILLING_CONTACT = "BillingContact";
+    private static final String SHIPPING_CONTACT = "ShippingContact";
+    private static final String IDENTIFIER = "Identifier";
+    private static final String TOKEN_DATA = "TokenData";
+
+    @NonNull public final MailingAddressInput billingContact;
+    @NonNull public final ShippingContact shippingContact;
+    @Nullable public final String identifier;
+    @NonNull public final TokenData tokenData;
+
+    private NativePayment(Builder builder) {
+        billingContact = builder.billingContact;
```

Comment:
Probably minor but what happens if we never set `billingContact` in the builder and we assign null to this non-null property? Does it make sense to add a layer of validation in the `build` method to make sure we don't try to do this?

Reply:
Of course it does. Will do it. :)

Reply:
https://github.com/Shopify/unity-buy-sdk/pull/356/commits/3b721d459bbdfcf562a5a2f0900809b1796330fd